src/pages/students/[student_id]/goals/[goal_id]/create.tsx

```diff
@@ -57,102 +153,115 @@ const CreateBenchmarkPage = () => {
         )}
       </Box>
 
-      <Divider />
+      <Box bgcolor={"var(--grey-80)"} py={4}>
+        <Typography variant="h3" textAlign="left" pb={2}>
+          Create Benchmark
+        </Typography>
+        <Stepper activeStep={viewState} alternativeLabel connector={null}>
+          {steps.map((label, index) => (
+            <Step key={label}>
+              {index !== steps.length && (
+                <StepLabel
+                  StepIconComponent={
+                    index < viewState ? CheckCircle : TripOriginRounded
+                  }
+                >
+                  {label}
+                </StepLabel>
+              )}
+            </Step>
+          ))}
+        </Stepper>
+      </Box>
 
       <fieldset disabled={addSubgoalMutation.isLoading} style={{ border: 0 }}>
-        <Stack
-          direction="row"
-          divider={<Divider orientation="vertical" flexItem />}
-          spacing={4}
-        >
-          <Stack spacing={4} p={3} pr={0}>
-            <Typography variant="h4">Benchmark</Typography>
-            <TextField
-              label="Description"
-              multiline
-              required
-              minRows={3}
-              name="description"
-            />
-            <TextField
-              label="Setup"
-              multiline
-              required
-              minRows={3}
-              name="setup"
-            />
-
-            <Stack spacing={2}>
-              <Typography variant="h4">Metrics</Typography>
-              <Stack direction="row" spacing={4}>
-                <Stack direction="row" alignItems="center">
+        <Stack direction="row" spacing={4}>
+          {viewState === VIEW_STATES.BENCHMARK_PG_1 && (
+            <Stack spacing={4} px={3} pb={3} width={"100%"}>
```

Comment:
This is a really cool approach, but it looks to me like when advancing to the second page, the content from VIEW_STATES.BENCHMARK_PG_1 is dropped, and then the data that filled out those text fields is lost. The thing I immediately think of to suggest is to render both "pages", but set visibility conditionally based on which page is selected. I don't know if that would cause an issue with FormData's ability to retrieve content, though. The other idea I would have would be to create a piece of state for each "page" that is set by collecting formData when clicking through to "Next" or to "Create Benchmark", then submitting the "page 1 state" and "page 2 state" together at the time of form submission. Just spitballing here, so feel free to do whatever works.

Reply:
After our conversation, it ended up that, since the form is split into two view states, plan B was the best option - using the state object data for the subgoalMutation, rather than the FormData API.